src/Cake.Core/Tooling/Tool.cs

```diff
@@ -226,6 +226,10 @@ protected IProcess RunProcess(TSettings settings, ProcessArgumentBuilder argumen
         /// <returns>The working directory for the tool.</returns>
         protected virtual DirectoryPath GetWorkingDirectory(TSettings settings)
         {
+            if (settings.WorkingDirectory != null)
+            {
+                return settings.WorkingDirectory;
+            }
             return _environment.WorkingDirectory;
```

Comment:
@patriksvensson is it just me that would like to see a new line after the closing bracket here?  If so, I will get over it :smile: